pennylane/hf/dipole.py

```diff
@@ -0,0 +1,401 @@
+        \hat{c}_{p\downarrow}^\dagger \hat{c}_{q\downarrow}] -
+        \hat{D}_\mathrm{c} + \hat{D}_\mathrm{n},
+
+    where the matrix elements :math:`d_{pq}` are given by the integral over molecular orbitals
+    :math:`\phi`
+
+    .. math::
+
+        d_{pq} = \int \phi_p^*(r) \hat{{\bf r}} \phi_q(r) dr,
+
+    :math:`\hat{c}^{\dagger}` and :math:`\hat{c}` are the creation and annihilation operators,
+    respectively, and :math:`\hat{D}_\mathrm{c}` and :math:`\hat{D}_\mathrm{n}` are the contribution
+    of the core orbitals and nuclei, respectively, which are computed as
+
+    .. math::
+
+        \hat{D}_\mathrm{c} = 2 \sum_{i=1}^{N_\mathrm{core}} d_{ii},
+
+    and
+
+    .. math::
+
+        \hat{D}_\mathrm{n} = \sum_{i=1}^{N_\mathrm{atoms}} Z_i {\bf R}_i \hat{I},
+
+    where :math:`Z_i` and :math:`{\bf R}_i` denote, respectively, the atomic number and the
+    nuclear coordinates of the :math:`i`-th atom of the molecule.
+
+    The fermonic dipole operator is then transformed to the qubit basis which gives
+
+    .. math::
+
+        \hat{D} = \sum_{j} c_j P_j,
+
+    where :math:`c_j` is a numerical coefficient and :math:`P_j` is a ternsor product of
+    single-qubit Pauli operators :math:`X, Y, Z, I`.
+
+    Args:
+        mol (Molecule): the molecule object
+        cutoff (float): cutoff value for discarding the negligible dipole moment integrals
+        core (list[int]): indices of the core orbitals
+        active (list[int]): indices of the active orbitals
+
+    Returns:
+        function: function that computes the qubit dipole moment observable
+
+    **Example**
+
+    >>> symbols  = ['H', 'H']
+    >>> geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]], requires_grad = False)
+    >>> alpha = np.array([[3.42525091, 0.62391373, 0.1688554],
+    >>>                   [3.42525091, 0.62391373, 0.1688554]], requires_grad=True)
+    >>> mol = qml.hf.Molecule(symbols, geometry, alpha=alpha)
+    >>> args = [alpha]
+    >>> dipole_moment(mol)(*args)[2].terms[1]
+    [PauliZ(wires=[0]),
+     PauliY(wires=[0]) @ PauliZ(wires=[1]) @ PauliY(wires=[2]),
+     PauliX(wires=[0]) @ PauliZ(wires=[1]) @ PauliX(wires=[2]),
+     PauliZ(wires=[1]),
+     PauliY(wires=[1]) @ PauliZ(wires=[2]) @ PauliY(wires=[3]),
+     PauliX(wires=[1]) @ PauliZ(wires=[2]) @ PauliX(wires=[3]),
+     PauliZ(wires=[2]),
+     PauliZ(wires=[3])]
+    """
+
+    def _dipole(*args):
+        r"""Compute the qubit dipole moment observable.
+
+        Args:
+            args (array[array[float]]): initial values of the differentiable parameters
+
+        Returns:
+            (list[Hamiltonian]): x, y and z components of the dipole moment observable
+        """
+        d = []
+        d_ferm = fermionic_dipole(mol, cutoff, core, active)(*args)
+        for i in d_ferm:
+            d.append(qubit_operator(i, cutoff=cutoff))
+
+        return d
+
+    return _dipole
+
+
+def fermionic_one(core_constant, integral, cutoff=1.0e-12):
```

Comment:
Maybe ``core_constant` could be shortened to `constant` ?
```suggestion
def fermionic_one(constant, integral, cutoff=1.0e-12):
```